env/env_posix.cc

```diff
@@ -115,6 +129,28 @@ int cloexec_flags(int flags, const EnvOptions* options) {
   return flags;
 }
 
+class PosixDynamicLibrary : public DynamicLibrary {
+public:
+  PosixDynamicLibrary(const std::string & name, void *handle)
+    :  name_(name), handle_(handle) {
+    }
+  ~PosixDynamicLibrary() {
+      dlclose(handle_);
+  }
+  
+  virtual FunctionPtr LoadSymbol(const std::string & sym_name) override {
+    void *symbol = dlsym(handle_, sym_name.c_str());
```

Comment:
I think we should check the error of `dlsym`. We currently return nullptr when there is an error, but loses the error information.